tests/tests/test-txpool-multiple.ts

```diff
@@ -0,0 +1,120 @@
+import { expect } from "chai";
+import { customWeb3Request } from "../util/providers";
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+import { createContract } from "../util/transactions";
+import { Transaction } from "web3-core";
+
+/*
+  At rpc-level, there is no interface for retrieving emulated pending transactions - emulated
+    transactions that exist in the Substrate's pending transaction pool. Instead they are added to a
+    shared collection (Mutex) with get/set locking to serve requests that ask for this transactions
+    information before they are included in a block.
+    We want to test that:
+      - We resolve multiple promises in parallel that will write in this collection on the rpc-side
+      - We resolve multiple promises in parallel that will read from this collection on the rpc-side
+      - We can get the final transaction data once it leaves the pending collection
+  */
+describeDevMoonbeam("Pending Pool - Multiple pending transactions", (context) => {
+  let txHashs: string[];
+
+  before("Setup: Sending 10 transactions", async function () {
+    txHashs = await Promise.all(
```

Comment:
```suggestion
    txHashes = await Promise.all(
```

Reply:
I fixed it after in the testweek branch directly